### ISSUE TITLE
Error "ansible-galaxy role init"

### ISSUE THREAD

**Issue by robertdebock** [ansible/ansible#71977]
##### SUMMARY

Since some time (likely since installing Ansible 2.10) the command `ansible-galaxy role init --role-skeleton=ansible-role-skeleton example` fails. This used to work but now returns:

```
ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: './example/.git/HEAD'
```

I'm using [this skeleton](https://github.com/robertdebock/ansible-role-skeleton) repository.

##### ISSUE TYPE

- Bug Report

##### COMPONENT NAME

ansible-galaxy

##### ANSIBLE VERSION
```
ansible 2.10.1
  config file = None
  configured module search path = ['/home/robertdb/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/robertdb/.local/lib/python3.8/site-packages/ansible
  executable location = /home/robertdb/.local/bin/ansible
  python version = 3.8.5 (default, Aug 12 2020, 00:00:00) [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

##### CONFIGURATION
```
# no output
```

##### OS / ENVIRONMENT

Running on Fedora 32, installed Ansible using pip.

##### STEPS TO REPRODUCE

```
git clone git@github.com:robertdebock/ansible-role-skeleton.git
ansible-galaxy role init --role-skeleton=ansible-role-skeleton/ example
```

##### EXPECTED RESULTS

I was hoping for a wonderfully initialized ansible role.

##### ACTUAL RESULTS

An error, unexpected. ;-)

```
ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: './example/.git/HEAD'

Traceback (most recent call last):
  File "/Users/matt/projects/ansibledev/ansible/bin/ansible-galaxy", line 125, in <module>
    exit_code = cli.run()
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/galaxy.py", line 499, in run
    context.CLIARGS['func']()
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/galaxy.py", line 934, in execute_init
    shutil.copyfile(os.path.join(root, f), os.path.join(obj_path, f_rel_path))
  File "/Users/matt/venvs/ansibledev/lib/python3.8/shutil.py", line 259, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: './example/.git/config'
```

**Comment by ansibot** [comment]
Files identified in the description:
* `lib/ansible/galaxy`

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Issue was introduced in 39b394204870fda6de789809a3d69f58da3e7c31

It looks like we need to mutate `dirs` instead of replacing it.

**Comment by robertdebock** [comment]
It could indeed be an easy fix indeed. As [hacktoberfest](https://hacktoberfest.digitalocean.com/) starts tomorrow and I need a t-shirt because all physical conferences are cancelled:

I'll make a PR tomorrow.

**Comment by manas-init** [comment]
Created a PR for the bug: https://github.com/ansible/ansible/pull/72035

**Comment by Praveenk8051** [comment]
Is this still open for first-time contributors?

**Comment by robertdebock** [comment]
Hey, I noticed this issue still occurs on Ansible 2.10.5. What's required to get PR #72035 merged?

Workaround:

```shell
mv skeleton/.git skeleton/.git.not
ansible-galaxy init --role-skeleton=skeleton MY_ROLE
# - Role MY_ROLE was created successfully
mv skeleton/.git.not/ skeleton/.git
```

**Comment by s-hertel** [comment]
It's a pattern that would be easy to mistakenly change again, so it's waiting on a test. Anyone who has time should feel free to cherry-pick the initial work to keep attribution and open a new PR with a test, since it appears the original author may not be interested in working on it anymore.

**Comment by pbpblush** [comment]
The error still occurs in 2.10.6
